### ISSUE TITLE
Use case 1: Multiple identifiers for a single bore

### ISSUE THREAD

**Issue by John-Sharples** [opengeospatial/boreholeie#16]
In Australia we run up against issues of multiple identification numbers/names for a single bore hole. For example, a state agency will assign a name  and an ID ((i.e. name=Burke flats 1, ID=1234). If this bore is incorporated into the national data set it is assigned another ID to ensure uniqueness across all states.

This can be compounded again when state level agencies merge/split or change data management technologies. In the past this has lead to renaming of bores to accommodate these changes, hence some bores have "historical IDs" as well as (multiple) current identifiers. 

Multiple pipes in a single bore hole can also impact IDs.

**Comment by denevers** [comment]
The ISO feature model can support multiple names (names are really identifiers), each scoped by their "codeSpace" (which provide the authority for the value).  In GML encoding it looks like (I just made up those examples)

`<gml:name codeSpace="http://www.nrcan.gc.ca/wells/nationalId">BH1967-00143</gml:name>
<gml:name codeSpace="https://mern.gouv.qc.ca">PUITS AB-010-1967</gml:name>`

it does not provide a validity time nor any lineage though (ie, this name was valid between 1971 and 1978 and then replace by this name).

Does this address this issue ?

**Comment by sgrellet** [comment]
"names are really identifiers" : kind of surprised with this. To me the identifier is the gml:identifier in which we could passe a URI

On the lineage, INSPIRE defined its own approach and defined an inspireID (including namespace and version) but this creates a mess to understand as it overlaps partially with gml:identifier.
Thus it is sometimes discussed to drop this see (https://inspire.ec.europa.eu/implementation-identifiers-using-uris-inspire-%E2%80%93-frequently-asked-questions/59309 -> search for  gml:identifier)

Back to John's issue : to handle this, in the French Water Information System, we have modelled an dedicated class for 'alternate identifiers' -> do we want this here ?

**Comment by denevers** [comment]
In short, your identifier is somebody else name

From GML spec

7.2.4.4 name, identifier

The gml:name property provides a label or identifier for the object, commonly a descriptive name.
An object may have several names, typically assigned by different authorities. gml:name uses the
gml:CodeType content model. The authority for a name is indicated by the value of its (optional) codeSpace attribute. The name may or may not be unique, as determined by the rules of the organization responsible for the codeSpace. In common usage there will be one name per authority, so a processing application may select the name from the codeSpace that it prefers.

`<element name="name" type="gml:CodeType"/>`

Often, a special identifier is assigned to an object by the authority that maintains the feature with the intention that it is used in references to the object. For such cases, the codeSpace shall be provided. That identifier is usually unique either globally or within an application domain. gml:identifier is a predefined property for such identifiers.

**Comment by John-Sharples** [comment]
Thanks Eric, I think that solves my issue.

I don't think the period of applicability needs to be included as these typically apply to changes in department name or managment system. Those details should be reflected in the codeSpace

**Comment by sgrellet** [comment]
"In short, your identifier is somebody else name". ok, I overlooked this part of the def

Back to this IE IT formalism, GML spec provide us with an answer, good
We haven't yet discussed the way we'll express/model our Borehole model. To me, GML (logical model) is not what we seek. For example, within OGC, based on what was done for GWML2, there was no GML artefact in the conceptual model.
If we take the conceptual model route (or set up a formal ontology), I still feel we need to model this multiple identifier issue explicitely -> add an extra 'externalIdentifier' property ? [0..*] in a conceptual model ?

**Comment by denevers** [comment]
right, I assumed the abstract ISO feature model, but we should not assume it is.
So,go fo 0..* external Identifiers

**Comment by sgrellet** [comment]
After 15/10/2018 webconf: two sub-aspects
- Keeping track of all names (can still be out there): historical variations (organization merging) -> one active identifier when you are doing the merge. 
- Several identifiers at the same time

Seems feasible to cover both at a time with a dedicated class 'external identifiers’ containing
- Identifier
- TimeSpan
- and also a potential source/context property (after discussion with @martluth )

**Comment by reynones** [comment]
Hello world!. 
I'm just starting with the subject. Here (Argentina) we use UWI code as the unique identifier but national authorities set the structure of Oil&Gas' well names. Now we face a problem with naming of multipad wells.   Have you opinion about that?

**Comment by sgrellet** [comment]
Should also be linked to #56 discussions

**Comment by sgrellet** [comment]
@reynones : thanks for your contributions.
Sorry not replying sooner I overlooked the notification and only go back here in the process of finalizing the Borehole IE Engineering Report.
We mainly focused on positioning elements (observation, construction, samples, ...) along the Trajectory.
I guess we'll move on those topics when the work is more formalized (ex : having an official Borehole Standard Working Group)
